packages/logger/src/Logger.ts

```diff
@@ -432,25 +454,24 @@ class Logger extends Utility implements LoggerInterface {
         } catch (error) {
           throw error;
         } finally {
-          Logger.injectLambdaContextAfterOrOnError(
-            loggerRef,
-            initialPersistentAttributes,
-            options
-          );
+          Logger.injectLambdaContextAfterOrOnError(loggerRef, {}, options);
         }
 
         return result;
       };
     };
   }
 
+  /**
+   * @deprecated This method is deprecated and will be removed in the future major versions.
```

Comment:
We're using `resetState()` directly - but I initially left the method usage in because otherwise I'd have to also change the tests.

But on second thought you're right - let's just rip the bandaid. I have removed the usage, changed the tests, and excluded the method from the test coverage (since it's no longer used it's not covered).